src/Authoring/WinRT.SourceGenerator/WinRTTypeWriter.cs

```diff
@@ -935,25 +935,29 @@ internal static (string, string, string, bool, bool) GetSystemTypeCustomMapping(
         private void ProcessCustomMappedInterfaces(INamedTypeSymbol classSymbol)
         {
             Logger.Log("writing custom mapped interfaces for " + QualifiedName(classSymbol));
+            Dictionary<INamedTypeSymbol, bool> isPublicImplementation = new Dictionary<INamedTypeSymbol, bool>();
+
             // Mark custom mapped interface members for removal later.
             // Note we want to also mark members from interfaces without mappings.
             foreach (var implementedInterface in GetInterfaces(classSymbol, true).
                 Where(symbol => MappedCSharpTypes.ContainsKey(QualifiedName(symbol)) ||
                                 ImplementedInterfacesWithoutMapping.Contains(QualifiedName(symbol))))
             {
-                string interfaceName = QualifiedName(implementedInterface);
-                Logger.Log("custom mapped interface: " + interfaceName);
+                bool isPubliclyImplemented = false;
+                Logger.Log("custom mapped interface: " + QualifiedName(implementedInterface, true));
                 foreach (var interfaceMember in implementedInterface.GetMembers())
                 {
                     var classMember = classSymbol.FindImplementationForInterfaceMember(interfaceMember);
                     currentTypeDeclaration.CustomMappedSymbols.Add(classMember);
+                    isPubliclyImplemented |= (classMember.DeclaredAccessibility == Accessibility.Public);

```

Comment:
if the first member is public, isPubliclyImplemented := true
if the second member is not public, isPubliclyImplemented is still true... Seems like we would want it to be false? so &=?

Reply:
Thats a good question and I probably should add a comment in the code there.  The issue with custom mapped interfaces is we do not have a 1 to 1 mapping between the members of the mapped from interface and the mapped to interface.  This means we can't say since these 2 members were explicitly implemented while the others weren't, we will find those 2 members in the mapped interface and mark them private.  We need to actually decide whether the entire interface is private (explicitly implemented) or not.  So the decision chosen here based on what I see .NET native doing is if at least one member is not explicitly implemented,  the entire mapped interface is treated as a public (not explicitly implemented).  Only if all the members were explicitly implemented is when we treat the mapped interface as also explicitly implemented.

Reply:
could break the loop as soon as isPubliclyImplemented is true

---
In reply to: [571209062](https://github.com/microsoft/CsWinRT/pull/720#discussion_r571209062) [](ancestors = 571209062)

Reply:
I won't be able to break out of the loop as I am using the loop to collect other information too (CustomMappedSymbols).  I can initialize the dictionary entry first outside of the loop to false and then do the |= in the loop and remove isPubliclyImplemented, but its almost equivalent to what is there with additional dictionary lookup.